src/parse.rs

```diff
@@ -2284,6 +2284,8 @@ impl<'a> Parser<'a> {
     /// precedence, because the URL of links must not be processed.
     fn handle_inline_pass1(&mut self) {
         let mut link_stack = Vec::new();
+        let mut codeblock_delims: Vec<(usize, TreeIndex)> = Vec::new();
```

Comment:
I guess this could also be a `HashMap<usize, TreeIndex>` where we store the index of the last occurrence of each length delimiter. And then a scan for a match would simply be a lookup in the table plus a check if any match is the same as the opening delimiter.

Reply:
I've thought about this more, and just keeping track of the last occurrence of each delimiter size won't work. This would get parsed incorrectly:
```
` first occ of double backtick: `` snd: `` and third: ``
```
As the first occurrence of the double back tick would incorrectly match the last instead of the second.

Reply:
I've thought about this a little and it's tricky. I think the general approach is to retain the results of the forward scan so it doesn't need to get redone.

An appropriate data structure is probably just Vec<Option<TreeIndex>>, where the index is the count. In real life, indices are tiny. In asymptotic analysis, this is O(n) which is acceptable, even if it *feels* like a sparser data structure should be better.

Reply:
I'm not sure if that that is much better than this PR's solution though (which also retains the forward scan results). It should be just as fast for common scenarios and also has O(n) worst-time complexity.